### PR TITLE
Support text inputs in the v2 server

### PR DESCRIPTION
**Pull request by EternityForest**
This PR adds simple support for the mauritskorse text components in the v2 server.  Also, the readme directions did not work, I assume they were typos, and changed them to match the current code.

mauritskorse's PR which adds the components

https://github.com/esphome/esphome/pull/3638

Tested against my forked branch of the text components, as the original appears to have not been updated
in a while and did not seem to like the current PlatformIO S2 toolchain.

https://github.com/EternityForest/esphome/tree/sync-add-text-input-component

### CODE REVIEW

File: README.md
Comment (jesserockz) [inline README.md:1]: Can you put these in their own PR please?

**Comment by EternityForest** [comment]
The Readme change and the inputs? Can do, should I just split them and
force push?

File: README.md
Comment (EternityForest) [reply README.md:1]: Done!

Comment (EternityForest) [reply README.md:1]: Oops, misread that as "their own commit"... it's been a long day!

**Review by jesserockz — APPROVED** [review]
